Modules/CPackDeb.cmake

```diff
@@ -423,6 +448,34 @@ if(NOT UNIX)
   message(FATAL_ERROR "CPackDeb.cmake may only be used under UNIX.")
 endif()
 
+function(get_component_package_name var component)
+  string(TOUPPER "${component}" component_upcase)
+  if(CPACK_DEBIAN_${component_upcase}_PACKAGE_NAME)
+    string(TOLOWER "${CPACK_DEBIAN_${component_upcase}_PACKAGE_NAME}" package_name)
+  else()
+    string(TOLOWER "${CPACK_PACKAGE_NAME}-${component}" package_name)
+  endif()
+
+  set("${var}" "${package_name}" PARENT_SCOPE)
+endfunction()
+
+#extract library name and version for given shared object
+function(extract_soname shared_object libname version)
+  if(READELF_EXECUTABLE)
+    execute_process(COMMAND "${READELF_EXECUTABLE}" -d "${shared_object}"
+      WORKING_DIRECTORY "${CPACK_TEMPORARY_DIRECTORY}"
+      RESULT_VARIABLE _result
+      OUTPUT_VARIABLE _output
+      ERROR_QUIET
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(${_result} EQUAL 0)
+      string(REGEX MATCH "Library soname: \\[([^\n]+)\\.so\\.([^\n]*)\\]\n" _soname "${_output}")
```

Comment:
I see a "slight" problem with that...

```
$ readelf -d /lib/x86_64-linux-gnu/libc.so.6 | grep SONAME
 0x000000000000000e (SONAME)             Библиотека soname: [libc.so.6]
```

Why not use the non-localizable `SONAME` string instead?

Also, why is `.so` in the middle of that regex? What if my library doesn't have `.so` in it?

Also, what if the match fails?